src/views/pages/api-test/TestPost.js

```diff
@@ -0,0 +1,136 @@
+            <CardHeader title="/test_post" subheader="Test POST API Endpoint" />
+            <Divider />
+            <CardContent component={Stack} spacing={2}>
+                <Paper variant="outlined" sx={{ p: 1 }}>
+                    <Typography variant="h4" p={2}>
+                        States:
+                    </Typography>
+                    <Box textAlign="center">
+                        <Chip label={"N1: " + calcState.n1} />
+                        <Chip label={"N2: " + calcState.n2} />
+                        <Chip label={"OP: " + calcState.op} />
+                    </Box>
+                </Paper>
+
+                <Paper variant="outlined" sx={{ p: 1 }}>
+                    <Typography variant="h4" p={2}>
+                        Inputs:
+                    </Typography>
+                    <Stack spacing={2} direction="row">
+                        <TextField
+                            required
+                            type="number"
+                            name="n1"
+                            label="n1"
+                            variant="outlined"
+                            onChange={handleTextInputChange}
+                            value={calcState.n1}
+                        />
+                        <Select
+                            labelId="op"
+                            name="op"
+                            value={calcState.op}
+                            label="Operation"
+                            onChange={handleTextInputChange}
+                            sx={{ width: 300 }}
+                        >
+                            <MenuItem value="add">Addition</MenuItem>
+                            <MenuItem value="sub">Subtraction</MenuItem>
+                            <MenuItem value="mul">Multiplication</MenuItem>
+                            <MenuItem value="div">Division</MenuItem>
+                            <MenuItem value="invalid">Invalid</MenuItem>
+                        </Select>
+                        <TextField
+                            required
+                            type="number"
+                            name="n2"
+                            label="n2"
+                            variant="outlined"
+                            onChange={handleTextInputChange}
+                            value={calcState.n2}
+                        />
+                    </Stack>
+                </Paper>
+                <Paper variant="outlined" sx={{ p: 1 }}>
+                    <Typography variant="h4" p={2}>
+                        Result:
+                    </Typography>
+                    <Stack justifyContent="center">
+                        <Item>
+                            <StateHandler apiState={calcApi} retryHandler={makeRequest}>
+                                <Chip label={calcState.n1 + " " + calcState.op + " " + calcState.n2 + " = " + calcApi.data} />
+                            </StateHandler>
```

Comment:
State handler is pretty important. Definitely look into how and why it works. It gets rid of lots of lines of code and handles a lot of stuff for you.